lib/src/configuration.dart

```diff
@@ -383,16 +383,64 @@ class Configuration {
     return pubspec;
   }
 
+  String? _getPubspecVersionWithBuildNumber(Version pubspecVersion) {
+    const prefix = "Warning: version_with_build_number is enabled,";
+    if (pubspecVersion.build.isNotEmpty && pubspecVersion.build[0] is int) {
+      final buildNumber = pubspecVersion.build[0] as int;
+      if (pubspecVersion.major > 65535) {
+        _logger.stderr(
+          '$prefix but the major version ${pubspecVersion.major} is higher than 65535.',
+        );
+        return null;
+      }
+      if (pubspecVersion.minor > 654) {
+        _logger.stderr(
+          '$prefix but the minor version ${pubspecVersion.minor} is higher than 654.',
+        );
+        return null;
+      }
+      if (pubspecVersion.patch > 99) {
+        _logger.stderr(
+          '$prefix but the patch version ${pubspecVersion.patch} is higher than 99.',
+        );
+        return null;
+      }
+      if (buildNumber > 65535) {
+        _logger.stderr(
+          '$prefix but the build version ${buildNumber} is higher than 65535.',
+        );
+        return null;
+      }
+      final combinedPatchNumber =
+          int.parse('${pubspecVersion.minor}0${pubspecVersion.patch}');
+      return [
+        pubspecVersion.major,
+        combinedPatchNumber,
+        buildNumber,
+        0,
```

Comment:
Then with above, go ahead and explicitly convert to string
```suggestion
      return [
        pubspecVersion.major.toString(),
        combinedPatchNumber,
        buildNumber.toString(),
        '0',
```